src/tickit/core/components/device_simulation.py

```diff
@@ -74,3 +80,14 @@ async def on_tick(self, time: SimTime, changes: Changes) -> None:
         )
         self.last_outputs = device_update.outputs
         await self.output(time, out_changes, device_update.call_at)
+
+    async def stop_component(self) -> None:
+        """Cancel all pending tasks associated with the device component.
+
+        Cancels long running adapter tasks associated with the component.
+        """
+        LOGGER.debug("Stopping {}".format(self.name))
+        if self._tasks:
+            for task in self._tasks:
+                task.cancel()
+            await asyncio.wait(self._tasks)
```

Comment:
The `if` here is superfluous:
```suggestion
        for task in self._tasks:
            task.cancel()
        await asyncio.wait(self._tasks)
```

Reply:
Same as system simulation.